In [35]:
%reset -f

In [92]:
import pandas as pd
import os
import numpy as np

In [39]:
current_directory = os.getcwd()
data_path = current_directory +"/../../DATA/"

csv1_name = 'sba_dataset.csv'
csv2_name = 'SBS_Certified_Business_List.csv'
csv3_name = 'subsectors_description_Kodiak.csv'

path1 = data_path + csv1_name
path2 = data_path + csv2_name
path3 = data_path + csv3_name

dat1 = pd.read_csv(path1)
dat2 = pd.read_csv(path2)
dat3 = pd.read_csv(path3)

In [38]:
def short_dataframe_by_number_of_nulls(df):

    null_counts = df.isnull().sum()
    sorted_columns = null_counts.sort_values()
    sorted_column_names = sorted_columns.index.tolist()
    df_shorted = df[sorted_column_names]
    return df_shorted

In [45]:
dat1 = short_dataframe_by_number_of_nulls(dat1)
dat2 = short_dataframe_by_number_of_nulls(dat2)
dat3 = short_dataframe_by_number_of_nulls(dat3)

In [67]:
dat1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65286 entries, 0 to 65285
Data columns (total 22 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   BUSINESS_NAME                  65286 non-null  object 
 1   FULL_ADDRESS                   65286 non-null  object 
 2   TS                             65286 non-null  object 
 3   CONTACT                        65285 non-null  object 
 4   YEAR_ESTABLISHED               65282 non-null  float64
 5   MAIN_OR_BRANCH                 65282 non-null  object 
 6   EMAIL_ADDRESS                  65281 non-null  object 
 7   NAICS                          65281 non-null  object 
 8   PHONE_NUMBER                   65256 non-null  object 
 9   LEGAL_STRUCTURE                64498 non-null  object 
 10  WEBPAGE                        27092 non-null  object 
 11  DESCRIPTION_KEYWORDS           23081 non-null  object 
 12  IS_EXPORTER                    17740 non-null 

- Some useful columns:

BUSINESS_NAME: Name of a the company. Probably the ID of the dataset

NAICS: The identification code for the company. This is the variable we want to predict. Note: there are multiple values for each company, and we don't know the reason yet.

YEAR_ESTABLISHED: When the company started

WEBPAGE*: The website of the company

EMALL_PAGE_URL*: Another website

DESCRIPTION_KEYWORDS*: Some key words about what the company does. This is not a summary of the column DESCRIPTION_OF_OPERATIONS, since the can have really different 

IS_EXPORTER*: Possible values: No, Yes, Wants to be

DESCRIPTION_OF_OPERATIONS*: This value doesn't match with  DESCRIPTION_KEYWORDS

BUSINESS_OPT_TYPE*: A classification of the business

SPECIAL_EQUIPMENT_MATERIALS**:  Maybe products or items the sell 

EXPORT_BUSINESS_ACTIVITIES**: This column has similar information than BUSINESS_OPT_TYPE when they are both not null. 

EXPORT_BUSINESS_RELATIONSHIPS**: To whom the company sell

EXPORT_OBJECTIVE**: A goal of the company     

EXPORT_COUNTRIES**: To which countries the company sell

QUALITY_ASSURANCE_STANDARDS**: Some  quality guidelines or standards.    


- Contact columns

FULL_ADDRESS: Address of the company

CONTACT: The contact person 

EMAIL_ADDRESS: The email of the contact person

PHONE_NUMBER: The phone number of the contact person 

- Don't understand columns

LEGAL_STRUCTURE: this column have the following values: LL, CP, SC, SL, PS.

MAIN_OR_BRANCH



NOTE:

Size of the dataset: 65286 

\* less than 30000 non_nulls

** less than 10000 non_nulls



In [124]:
df_doo = dat1[['DESCRIPTION_OF_OPERATIONS','NAICS']]
df_doo_not_null = df_doo[df_doo['DESCRIPTION_OF_OPERATIONS'].notnull() & df_doo['NAICS'].notnull()]


In [140]:
n = len(df_doo_not_null['NAICS'])
unique_NAICS = np.full(n, False, dtype=bool)

for i in range(n):
    
    if (len(df_doo_not_null['NAICS'].iloc[i]) == 6):
        unique_NAICS[i] = True

In [143]:
np.unique(unique_NAICS, return_counts=True)

(array([False,  True]), array([13821,  2757]))

In [95]:
len(dat1['NAICS'])

65286

'sba_dataset.csv'
- Number of observations: 65286
- Number of observations with DESCRIPTION_OF_OPERATIONS=!null and unique NAICS: 2757

In [151]:
df_doo_null_and_dk_not_nulls = dat1[dat1['DESCRIPTION_OF_OPERATIONS'].isnull() & dat1['NAICS'].notnull() & dat1['DESCRIPTION_KEYWORDS'].notnull()]
len(df_doo_null_and_dk_not_nulls)

6505

Potential new rows to be analyzed with 
- DESCRIPTION_OF_OPERATIONS==null 
- DESCRIPTION_KEYWORDS=!null
DESCRIPTION_KEYWORDS could ben an interesting value